tests/provider: AMI Data Source tests failing (GovCloud)

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fail on GovCloud:

```
TestAccAWSAmiDataSource_instanceStore
TestAccAWSAmiDataSource_localNameFilter
TestAccAWSAmiDataSource_natInstance
TestAccAWSAmiDataSource_windowsInstance
```

With related errors: `Your query returned no results. Please change your search criteria and try again.` or `Attribute 'owner_id' expected "801119661308", got "077303321853"`.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* data.aws_ami

### Terraform Configuration Files

Example config:

```hcl
data "aws_ami" "nat_ami" {
  most_recent = true
  owners      = ["amazon"]

  filter {
    name   = "name"
    values = ["amzn-ami-vpc-nat*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  filter {
    name   = "root-device-type"
    values = ["ebs"]
  }

  filter {
    name   = "block-device-mapping.volume-type"
    values = ["standard"]
  }
}
```

### Debug Output

```
TestAccAWSAmiDataSource_localNameFilter: data_source_aws_ami_test.go:135: Step 1/1 error: Error running pre-apply refresh: 2020/10/28 05:31:35 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Your query returned no results. Please change your search criteria and try again.

--- FAIL: TestAccAWSAmiDataSource_localNameFilter (9.83s)
```

Or

```
TestAccAWSAmiDataSource_natInstance: data_source_aws_ami_test.go:14: Step 1/1 error: Check failed: Check 14/25 error: data.aws_ami.nat_ami: Attribute 'owner_id' expected "137112412989", got "045324592363"
--- FAIL: TestAccAWSAmiDataSource_natInstance (28.12s)
```

### References

* #0000

--- a/aws/data_source_aws_ami_test.go
+++ b/aws/data_source_aws_ami_test.go
@@ -38,7 +38,7 @@ func TestAccAWSAmiDataSource_natInstance(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "image_type", "machine"),
 					resource.TestCheckResourceAttr(resourceName, "most_recent", "true"),
 					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("^amzn-ami-vpc-nat")),
-					resource.TestCheckResourceAttr(resourceName, "owner_id", "137112412989"),
+					testAccMatchResourceAttrAccountID(resourceName, "owner_id"),
 					resource.TestCheckResourceAttr(resourceName, "public", "true"),
 					resource.TestCheckResourceAttr(resourceName, "product_codes.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "root_device_name", "/dev/xvda"),
@@ -76,7 +76,7 @@ func TestAccAWSAmiDataSource_windowsInstance(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "image_type", "machine"),
 					resource.TestCheckResourceAttr(resourceName, "most_recent", "true"),
 					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("^Windows_Server-2012-R2")),
-					resource.TestCheckResourceAttr(resourceName, "owner_id", "801119661308"),
+					testAccMatchResourceAttrAccountID(resourceName, "owner_id"),
 					resource.TestCheckResourceAttr(resourceName, "platform", "windows"),
 					resource.TestCheckResourceAttr(resourceName, "public", "true"),
 					resource.TestCheckResourceAttr(resourceName, "product_codes.#", "0"),
@@ -96,13 +96,13 @@ func TestAccAWSAmiDataSource_windowsInstance(t *testing.T) {
 }
 
 func TestAccAWSAmiDataSource_instanceStore(t *testing.T) {
-	resourceName := "data.aws_ami.instance_store_ami"
+	resourceName := "data.aws_ami.amzn-ami-minimal-hvm-instance-store"
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:  func() { testAccPreCheck(t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckAwsAmiDataSourceInstanceStoreConfig,
+				Config: testAccLatestAmazonLinuxHvmInstanceStoreAmiConfig(),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAwsAmiDataSourceID(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "architecture", "x86_64"),
@@ -110,11 +110,11 @@ func TestAccAWSAmiDataSource_instanceStore(t *testing.T) {
 					resource.TestMatchResourceAttr(resourceName, "creation_date", regexp.MustCompile("^20[0-9]{2}-")),
 					resource.TestCheckResourceAttr(resourceName, "hypervisor", "xen"),
 					resource.TestMatchResourceAttr(resourceName, "image_id", regexp.MustCompile("^ami-")),
-					resource.TestMatchResourceAttr(resourceName, "image_location", regexp.MustCompile("ubuntu-trusty-14.04-amd64-server")),
+					resource.TestMatchResourceAttr(resourceName, "image_location", regexp.MustCompile("amzn-ami-minimal-hvm")),
 					resource.TestCheckResourceAttr(resourceName, "image_type", "machine"),
 					resource.TestCheckResourceAttr(resourceName, "most_recent", "true"),
-					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("^ubuntu/images/hvm-instance/ubuntu-trusty-14.04-amd64-server")),
-					resource.TestCheckResourceAttr(resourceName, "owner_id", "099720109477"),
+					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("amzn-ami-minimal-hvm")),
+					testAccMatchResourceAttrAccountID(resourceName, "owner_id"),
 					resource.TestCheckResourceAttr(resourceName, "public", "true"),
 					resource.TestCheckResourceAttr(resourceName, "product_codes.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "root_device_type", "instance-store"),
@@ -220,29 +220,6 @@ data "aws_ami" "windows_ami" {
 }
 `
 
-// Instance store test - using Ubuntu images
-const testAccCheckAwsAmiDataSourceInstanceStoreConfig = `
-data "aws_ami" "instance_store_ami" {
-  most_recent = true
-  owners      = ["099720109477"]
-
-  filter {
-    name   = "name"
-    values = ["ubuntu/images/hvm-instance/ubuntu-trusty-14.04-amd64-server*"]
-  }
-
-  filter {
-    name   = "virtualization-type"
-    values = ["hvm"]
-  }
-
-  filter {
-    name   = "root-device-type"
-    values = ["instance-store"]
-  }
-}
-`
-
 // Testing name_regex parameter
 const testAccCheckAwsAmiDataSourceNameRegexConfig = `
 data "aws_ami" "name_regex_filtered_ami" {
@@ -254,6 +231,6 @@ data "aws_ami" "name_regex_filtered_ami" {
     values = ["amzn-ami-*"]
   }
 
-  name_regex = "^amzn-ami-\\d{3}[5].*-ecs-optimized"
+  name_regex = "^amzn-ami-min[a-z]{4}-hvm"
 }
 `

--- a/aws/provider_test.go
+++ b/aws/provider_test.go
@@ -292,6 +292,13 @@ func testAccCheckResourceAttrHostnameWithPort(resourceName, attributeName, servi
 	}
 }
 
+// testAccMatchResourceAttrAccountID ensures the Terraform state regexp matches an account ID
+func testAccMatchResourceAttrAccountID(resourceName, attributeName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		return resource.TestMatchResourceAttr(resourceName, attributeName, regexp.MustCompile(`^\d{12}$`))(s)
+	}
+}
+
 // testAccMatchResourceAttrRegionalARN ensures the Terraform state regexp matches a formatted ARN with region
 func testAccMatchResourceAttrRegionalARN(resourceName, attributeName, arnService string, arnResourceRegexp *regexp.Regexp) resource.TestCheckFunc {
 	return func(s *terraform.State) error {

